packages/js/product-editor/src/blocks/product-fields/password/edit.tsx

```diff
@@ -32,26 +34,31 @@ export function Edit( {
 		'post_password'
 	);
 
-	const [ checked, setChecked ] = useState( Boolean( postPassword ) );
 	const postPasswordId = useInstanceId(
 		BaseControl,
 		'post_password'
 	) as string;
 
+	const { requirePassword } = useDispatch( productEditorUiStore );
+
+	const isPasswordRequired: boolean = useSelect( ( select ) => {
+		return select( productEditorUiStore ).isPasswordRequired();
+	}, [] );
```

Comment:
I mentioned this in a comment further below, but I am not sure on the reasoning behind this change and reliance on the new data store. This currently breaks the form and adds an unnecessary layer, I think we should revert these changes and remove the changes in the productEditorUiStore.